library/MedEx/API.php

```diff
@@ -0,0 +1,2092 @@
+                    // Yes, so do this only for recalls that are in the past!  T
+                   
+                    if (strtotime($recall['r_eventDate']) < mktime(0,0,0)) {
+                        if ($this->recursive_array_search("recall_".$recall['r_pid'],$appt3)) { continue; }
+                    }
+                     $count_recalls++;
+                    $recall2 = array();                    
+                    $recall2['pc_pid']        = $recall['r_pid'];
+                    $recall2['e_C_UID']       = $event['C_UID'];
+                    $recall2['pc_eventDate']  = $recall['r_eventDate'];
+                    $recall2['pc_startTime']  = '10:42:00';
+                    $recall2['pc_eid']        = "recall_".$recall['r_pid'];
+                    $recall2['e_pc_eid']      = "recall_".$recall['r_pid'];
+                    $recall2['pc_aid']        = $recall['r_provider'];
+                    $recall2['e_reason']      = (!empty($recall['r_reason']))?:'';
+                    $recall2['e_is_subEvent_of']= "0";
+                    $recall2['language']      = $recall['language'];
+                    $recall2['pc_facility']   = $recall['r_facility'];
+                    $recall2['fname']         = $recall['fname'];
+                    $recall2['lname']         = $recall['lname'];
+                    $recall2['mname']         = $recall['mname'];
+                    $recall2['street']        = $recall['street'];
+                    $recall2['postal_code']   = $recall['postal_code'];
+                    $recall2['city']          = $recall['city'];
+                    $recall2['state']         = $recall['state'];
+                    $recall2['country_code']  = $recall['country_code'];
+                    $recall2['phone_home']    = $recall['phone_home'];
+                    $recall2['phone_cell']    = $recall['phone_cell'];
+                    $recall2['email']         = $recall['email'];
+                    $recall2['C_UID']         = $event['C_UID'];
+                    $recall2['E_fire_time']   = $event['E_fire_time'];
+                    $recall2['time_order']    = $event['time_order'];
+                    $recall2['M_type']        = $event['M_type'];
+                    $recall2['reply']         = "To Send";
+                    $recall2['extra']         = "QUEUED";
+                    $recall2['status']        = "SENT";
+                    $recall2['to']            = $results;
+                    
+                    $appt3[] = $recall2;
+                }
+            }   
+        }
+        if (!empty($RECALLS_completed)) {
+            $hipaa = $this->process_deletes($token,$RECALLS_completed);
+        }
+        if (!empty($appt3)) {
+            $hipaa = $this->process($token,$appt3);
+        }     
+
+        $responses['count_appts'] = $count_appts;
+        $responses['count_recalls'] = $count_recalls;
+        return $responses;
```

Comment:
If you are referring to Patient messaging preferences, MedEx does refer to the messaging preferences in the demographics.  In fact MedEx relies completely on them to decide if a specific message type can be sent or is allowed to be sent on a per-patient level. 

I also noticed in the latest codebase there is a whole bunch of messaging options/rules - whatever the reasons for all the messaging settings, they are nothing short of confusing to average people who need to use the software.  I ignored them as much as possible.